python_fundamentals-master/07_classes_objects_methods/07_01_car.py

```diff
@@ -10,20 +10,25 @@
 
 '''
 
-class car():
-    def __init__(self, model, year, max_speed):
+class Car():
+    def __init__(self, model, year, speed):
         self.model = model
         self.year = year
-        self.max_speed = int(max_speed)
-
-    def speed_up(self, speed):
-        print("max_speed is " + (int(speed) + int(self.max_speed))
-
-smart = car("smart", "1993", 210)
-print(smart.model)
-print(smart.year)
-print(smart.speed_up(5))
-volvo = car("volvo", "2010", 260)
-print(volvo.model)
-print(volvo.year)
-print(volvo.speed_up(5))
+        self.speed = speed
+    def speed_up(self, speed_up):
+        self.speed_up = (self.speed + 5)
+        return f"max_speed is {self.speed_up}"
```

Comment:
Perfect, this actually modifies the `self.speed` attribute so the change would persist.

Reply:
When I run the code, it gives me the following error:
    print(car_1.speed_up(210))
TypeError: 'int' object is not callable

Reply:
yes because you are overwriting the function name on line 19
The name of the method is `speed_up` but then you set on line 19
`self.speed_up = int` !! then will overwrite the method and now `speed_up` is an integer, not a method anymore.
I would guess it works the first time (line 29) but not the second time (line 30)